src/Pages/homePage.tsx

```diff
@@ -6,7 +6,8 @@ const HomePage = () => {
     return (
         <div>
             <h1>Home Page</h1>
-            <h2>Welcome {location.state.username}</h2>
+            <h2 style={{color: 'green'}}>{location.state.message}</h2>
+            <h2>Welcome {location.state.email}</h2>
```

Comment:
@ShiranthaKellum, it's better if we destructure the `location.state` to extract username, message, and email values, refer below example,

```
// you can destructure values like this and you can add custom types as well 
const { username, message, email }: StateType = location?.state;

return (
    <div>
            <h2>Welcome {username}</h2>
            <h2 style={{color: 'green'}}>{message}</h2>
            <h2>Welcome {email}</h2>
    </div>
)
```